api/src/main/java/com/vinumeris/updatefx/UpdateFX.java

```diff
@@ -85,9 +85,17 @@ public static Path findAppExecutable(Path appInstallDir) throws IOException {
         if (os.contains("mac")) {
             // Returns path to the .app directory, which normally contains only a Contents directory.
             List<Path> exes = Utils.listDir(appInstallDir.resolve("Contents/MacOS"));
-            if (exes.size() != 1)
-                throw new IllegalStateException("Found unknown number of app executables");
-            return exes.get(0);
+            for (Path exe : exes) {
+                /**
+                 * Since 8u40, the packager introduced a "libpackager.dylib"
```

Comment:
I don't understand. We are talking about the mac case here right?
You're suggesting I only take the file that do not have extension ?